napalm_eos/eos.py

```diff
@@ -1003,120 +1003,133 @@ def get_mac_address_table(self):
-                nexthop_interface_map = {}
-                for next_hop in route_details.get('vias'):
-                    nexthop_ip = napalm_base.helpers.ip(next_hop.get('nexthopAddr'))
-                    nexthop_interface_map[nexthop_ip] = next_hop.get('interface')
-                metric = route_details.get('metric')
-                command = 'show ip{ipv} bgp {destination} detail'.format(
-                    ipv=ipv,
-                    destination=prefix
-                )
-                default_vrf_details = self.device.run_commands([command])[0].get(
-                    'vrfs', {}).get('default', {})
-                local_as = default_vrf_details.get('asn')
-                bgp_routes = default_vrf_details.get(
-                    'bgpRouteEntries', {}).get(prefix, {}).get('bgpRoutePaths', [])
-                for bgp_route_details in bgp_routes:
-                    bgp_route = route.copy()
-                    as_path = bgp_route_details.get('asPathEntry', {}).get('asPath', u'')
-                    remote_as = int(as_path.split()[-1])
-                    remote_address = napalm_base.helpers.ip(bgp_route_details.get(
-                        'routeDetail', {}).get('peerEntry', {}).get('peerAddr', ''))
-                    local_preference = bgp_route_details.get('localPreference')
-                    next_hop = napalm_base.helpers.ip(bgp_route_details.get('nextHop'))
-                    active_route = bgp_route_details.get('routeType', {}).get('active', False)
-                    last_active = active_route  # should find smth better
-                    communities = bgp_route_details.get('routeDetail', {}).get('communityList', [])
-                    preference2 = bgp_route_details.get('weight')
-                    inactive_reason = bgp_route_details.get('reasonNotBestpath', '')
-                    bgp_route.update({
-                        'current_active': active_route,
-                        'inactive_reason': inactive_reason,
-                        'last_active': last_active,
-                        'next_hop': next_hop,
-                        'outgoing_interface': nexthop_interface_map.get(next_hop),
-                        'selected_next_hop': active_route,
-                        'protocol_attributes': {
-                            'metric': metric,
-                            'as_path': as_path,
-                            'local_preference': local_preference,
-                            'local_as': local_as,
-                            'remote_as': remote_as,
-                            'remote_address': remote_address,
-                            'preference2': preference2,
-                            'communities': communities
-                        }
-                    })
-                    routes[prefix].append(bgp_route)
+            command = 'show ip{ipv} route vrf {_vrf} {destination} {protocol} detail'.format(
```

Comment:
Please add this command in a list of commands, then execute all of them in a single request.
Then iterate through the complete reply.